lib/serviceClient.ts

```diff
@@ -40,6 +41,14 @@ export interface ProxySettings {
   password?: string;
 }
 
+/**
+ * HTTP and HTTPS agents (Node.js only)
+ */
+export interface AgentSettings {
+  http: Agent;
+  https: Agent;
```

Comment:
Since these are typed by node's `http` module, I do wonder what happens if you try to TS compile against us without depending on the node lib.d.ts.

Reply:
As far as I can tell, compiling for the browser still works. Declaration files aren't used in a browser context and AFAIK the emitted Javascript drops imports for types and interfaces.

Is there a CI test for browser use? I ran `npm run test:karma:debugff` and got a few test failures, though they all seem unrelated. The failures are included below.

```
Firefox 82.0.0 (Mac OS X 10.15.0) defaultHttpClient should send HTTP requests FAILED
	Failed to send request to https://example.com
	RestError@test/msRest.browser.test.js:2096:28
	rejectOnTerminalEvent/<@test/msRest.browser.test.js:4981:63
	./node_modules/xhr-mock/lib/MockEventTarget.js/MockEventTarget.prototype.dispatchEvent/<@test/msRest.browser.test.js:98754:26
	./node_modules/xhr-mock/lib/MockEventTarget.js/MockEventTarget.prototype.dispatchEvent@test/msRest.browser.test.js:98752:36
	./node_modules/xhr-mock/lib/MockXMLHttpRequest.js/MockXMLHttpRequest.prototype.reportError@test/msRest.browser.test.js:99495:14
	./node_modules/xhr-mock/lib/MockXMLHttpRequest.js/MockXMLHttpRequest.prototype.handleError@test/msRest.browser.test.js:99598:14
	./node_modules/xhr-mock/lib/MockXMLHttpRequest.js/MockXMLHttpRequest.prototype.sendAsync/</<@test/msRest.browser.test.js:99449:30
	step@test/msRest.browser.test.js:99111:23
	verb/<@test/msRest.browser.test.js:99092:53
	rejected@test/msRest.browser.test.js:99084:57

Firefox 82.0.0 (Mac OS X 10.15.0) MsRestUserAgentPolicy for browser MsRestUserAgentPolicy (Browser) should be space delimited and contain two fields FAILED
	expected 5 to equal 2
	AssertionError@test/msRest.browser.test.js:7531:13
	./node_modules/chai/lib/chai/assertion.js/module.exports/Assertion.prototype.assert@test/msRest.browser.test.js:8585:13
	assertEqual@test/msRest.browser.test.js:9751:12
	methodWrapper@test/msRest.browser.test.js:16242:25
	./test/msRestUserAgentPolicyTests.ts/</</</</</<@test/msRest.browser.test.js:102162:61
	step@test/msRest.browser.test.js:93842:23
	verb/<@test/msRest.browser.test.js:93823:53
	fulfilled@test/msRest.browser.test.js:93813:58

Firefox 82.0.0 (Mac OS X 10.15.0) msrest serialize should correctly serialize a string in DateTimeRfc1123 FAILED
	expected 'Mon, 01 Jan 0001 00:00:00 GMT' to equal 'Mon, 01 Jan 2001 00:00:00 GMT'
	AssertionError@test/msRest.browser.test.js:7531:13
	./node_modules/chai/lib/chai/assertion.js/module.exports/Assertion.prototype.assert@test/msRest.browser.test.js:8585:13
	assertEqual@test/msRest.browser.test.js:9751:12
	methodWrapper@test/msRest.browser.test.js:16242:25
	./test/serializationTests.ts/</</</<@test/msRest.browser.test.js:103581:41

Firefox 82.0.0 (Mac OS X 10.15.0) XML serializer parseXML(string) with undefined FAILED
	error.message ("XML Parsing Error: syntax error
	Location: http://localhost:9876/context.html
	Line Number 1, Column 1:<sourcetext xmlns="http://www.mozilla.org/newlayout/xml/parsererror.xml">undefined
	^</sourcetext>") should have contained "Document is empty": expected -1 to not equal -1
	AssertionError@test/msRest.browser.test.js:7531:13
	./node_modules/chai/lib/chai/assertion.js/module.exports/Assertion.prototype.assert@test/msRest.browser.test.js:8585:13
	assertEqual@test/msRest.browser.test.js:9751:12
	methodWrapper@test/msRest.browser.test.js:16242:25
	./node_modules/chai/lib/chai/interface/assert.js/module.exports/assert.notStrictEqual@test/msRest.browser.test.js:12752:65
	./test/xmlTests.ts/</</</</<@test/msRest.browser.test.js:107133:73
	step@test/msRest.browser.test.js:93842:23
	verb/<@test/msRest.browser.test.js:93823:53
	fulfilled@test/msRest.browser.test.js:93813:58

Firefox 82.0.0 (Mac OS X 10.15.0) XML serializer parseXML(string) with null FAILED
	error.message ("XML Parsing Error: syntax error
	Location: http://localhost:9876/context.html
	Line Number 1, Column 1:<sourcetext xmlns="http://www.mozilla.org/newlayout/xml/parsererror.xml">null
	^</sourcetext>") should have contained "Document is empty": expected -1 to not equal -1
	AssertionError@test/msRest.browser.test.js:7531:13
	./node_modules/chai/lib/chai/assertion.js/module.exports/Assertion.prototype.assert@test/msRest.browser.test.js:8585:13
	assertEqual@test/msRest.browser.test.js:9751:12
	methodWrapper@test/msRest.browser.test.js:16242:25
	./node_modules/chai/lib/chai/interface/assert.js/module.exports/assert.notStrictEqual@test/msRest.browser.test.js:12752:65
	./test/xmlTests.ts/</</</</<@test/msRest.browser.test.js:107147:73
	step@test/msRest.browser.test.js:93842:23
	verb/<@test/msRest.browser.test.js:93823:53
	fulfilled@test/msRest.browser.test.js:93813:58

Firefox 82.0.0 (Mac OS X 10.15.0): Executed 347 of 375 (5 FAILED) (skipped 28) (0 secs / 1.255 secs)
Firefox 82.0.0 (Mac OS X 10.15.0): Executed 347 of 375 (5 FAILED) (skipped 28) (20.644 secs / 1.255 secs)
TOTAL: 5 FAILED, 342 SUCCESS
```

Reply:
Yeah, I think it'll be fine. I remember we had to fix some FireFox issues in `core-http` tests a while back. I believe @jeremymeng did the work if he remembers the PR.

Reply:
Yes the Firefox failures was fixed in core-http in https://github.com/Azure/azure-sdk-for-js/pull/9091/files.  Though the first failure in the above list seems new to me.

Reply:
If it passed in Chrome that is probably fine.

Reply:
I can confirm the tests pass on Chrome.